rubrics/impl/src/main/java/org/sakaiproject/rubrics/impl/RubricsServiceImpl.java

```diff
@@ -1522,6 +1523,181 @@ public Map<String, String> transferCopyEntities(String fromContext, String toCon
+
+            Element rubricEl = doc.createElement("rubric");
+            rubrics.appendChild(rubricEl);
+
+            rubricEl.setAttribute("title", rubric.getTitle());
+            rubricEl.setAttribute("created", Long.toString(rubric.getCreated().getEpochSecond()));
+            rubricEl.setAttribute("creator", rubric.getCreatorId());
+            rubricEl.setAttribute("weighted", Boolean.toString(rubric.getWeighted()));
+            rubricEl.setAttribute("adhoc", Boolean.toString(rubric.getAdhoc()));
+            rubricEl.setAttribute("max-points", Double.toString(rubric.getMaxPoints()));
+
+            Element criteriaEl = doc.createElement("criteria");
+            rubricEl.appendChild(criteriaEl);
+
+            rubric.getCriteria().forEach(criterion -> {
+
+                Element criterionEl = doc.createElement("criterion");
+                criteriaEl.appendChild(criterionEl);
+
+                criterionEl.setAttribute("title", criterion.getTitle());
+                Float weight = criterion.getWeight();
+                if (weight != null) {
+                    criterionEl.setAttribute("weight", Float.toString(weight));
+                }
+
+                String description = criterion.getDescription();
+                if (StringUtils.isNotBlank(description)) {
+                    Element descriptionEl = doc.createElement("description");
+                    criterionEl.appendChild(descriptionEl);
+                    descriptionEl.appendChild(doc.createCDATASection(description));
+                }
+
+                Element ratingsEl = doc.createElement("ratings");
+                criterionEl.appendChild(ratingsEl);
+
+                criterion.getRatings().forEach(rating -> {
+
+                    Element ratingEl = doc.createElement("rating");
+                    ratingsEl.appendChild(ratingEl);
+                    ratingEl.setAttribute("title", rating.getTitle());
+                    ratingEl.setAttribute("points", Double.toString(rating.getPoints()));
+                    String ratingDescription = rating.getDescription();
+                    if (StringUtils.isNotBlank(ratingDescription)) {
+                        Element ratingDescriptionEl = doc.createElement("description");
+                        ratingEl.appendChild(ratingDescriptionEl);
+                        ratingDescriptionEl.appendChild(doc.createCDATASection(ratingDescription));
+                    }
+                });
+            });
+        });
+
+        stack.pop();
+
+        results.append("completed archiving ").append(getLabel()).append(" for site ").append(siteId).append(LINE_SEPARATOR);
+        return results.toString();
+    }
+
+    @Override
+    public String merge(String toSiteId, Element root, String archivePath, String fromSiteId, Map<String, String> attachmentNames, Map<String, String> userIdTrans, Set<String> userListAllowImport) {
```

Comment:
One thing I am thinking about is teaching these routines to handle multiple identical imports.  They need to ignore the second and later imports of the same object.  Canvas does this.  For the site code, I did it by title.  Before I started walking the element, I grabbed a list of the tools already in the site and then whilst I was looping through the element, just skip those that match.  I can see issues with to skip or not to skip.  But I think that de-duping is the better approach.

Reply:
I can do that, no problem. Makes sense .. but which one wins? What if one is newer and the user wants to use that one?

Reply:
we should probably align the ui with import from site, using the same append or replace content approach. Just line them up. Also, we should take cues from OS semantics. So "The thing you're replacing is newer that the thing you want to replace it with, do you really want to?"